[2025-07-24T15:23:10.477+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-24T15:23:10.630+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: tunestream_etl_pipeline.Stage_songs scheduled__2025-07-24T02:00:00+00:00 [queued]>
[2025-07-24T15:23:10.671+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: tunestream_etl_pipeline.Stage_songs scheduled__2025-07-24T02:00:00+00:00 [queued]>
[2025-07-24T15:23:10.678+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 4
[2025-07-24T15:23:10.961+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): Stage_songs> on 2025-07-24 02:00:00+00:00
[2025-07-24T15:23:11.042+0000] {standard_task_runner.py:63} INFO - Started process 123 to run task
[2025-07-24T15:23:11.080+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'tunestream_etl_pipeline', 'Stage_songs', 'scheduled__2025-07-24T02:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/final_exam_src.py', '--cfg-path', '/tmp/tmpqdswp059']
[2025-07-24T15:23:11.117+0000] {standard_task_runner.py:91} INFO - Job 8: Subtask Stage_songs
[2025-07-24T15:23:11.478+0000] {task_command.py:426} INFO - Running <TaskInstance: tunestream_etl_pipeline.Stage_songs scheduled__2025-07-24T02:00:00+00:00 [running]> on host 12caace40239
[2025-07-24T15:23:12.078+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='tunestream_etl_pipeline' AIRFLOW_CTX_TASK_ID='Stage_songs' AIRFLOW_CTX_EXECUTION_DATE='2025-07-24T02:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-24T02:00:00+00:00'
[2025-07-24T15:23:12.091+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-24T15:23:12.514+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-24T15:23:12.597+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/final_exam_src.py", line 81, in load_songs_into_table
    postgres_hook.run(insert_sql, parameters=(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/common/sql/hooks/sql.py", line 431, in run
    with self._create_autocommit_connection(autocommit) as conn:
  File "/usr/local/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/common/sql/hooks/sql.py", line 551, in _create_autocommit_connection
    with closing(self.get_conn()) as conn:
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/postgres/hooks/postgres.py", line 145, in get_conn
    conn = deepcopy(self.connection or self.get_connection(conn_id))
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/hooks/base.py", line 83, in get_connection
    conn = Connection.get_connection_from_secrets(conn_id)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/connection.py", line 519, in get_connection_from_secrets
    raise AirflowNotFoundException(f"The conn_id `{conn_id}` isn't defined")
airflow.exceptions.AirflowNotFoundException: The conn_id `postgres_default` isn't defined
[2025-07-24T15:23:12.655+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=tunestream_etl_pipeline, task_id=Stage_songs, run_id=scheduled__2025-07-24T02:00:00+00:00, execution_date=20250724T020000, start_date=20250724T152310, end_date=20250724T152312
[2025-07-24T15:23:12.766+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 8 for task Stage_songs (The conn_id `postgres_default` isn't defined; 123)
[2025-07-24T15:23:12.896+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-24T15:23:13.001+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.10/site-packages/***/models/baseoperator.py:1297: AirflowProviderDeprecationWarning: Call to deprecated class PostgresOperator. (Please use `***.providers.common.sql.operators.sql.SQLExecuteQueryOperator`.Also, you can provide `hook_params={'schema': <database>}`.)
  result = cls.__new__(cls)

[2025-07-24T15:23:13.122+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-24T15:23:13.140+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-24T15:35:11.911+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-24T15:35:12.048+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: tunestream_etl_pipeline.Stage_songs scheduled__2025-07-24T02:00:00+00:00 [queued]>
[2025-07-24T15:35:12.107+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: tunestream_etl_pipeline.Stage_songs scheduled__2025-07-24T02:00:00+00:00 [queued]>
[2025-07-24T15:35:12.169+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 4
[2025-07-24T15:35:12.274+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): Stage_songs> on 2025-07-24 02:00:00+00:00
[2025-07-24T15:35:12.329+0000] {standard_task_runner.py:63} INFO - Started process 106 to run task
[2025-07-24T15:35:12.366+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'tunestream_etl_pipeline', 'Stage_songs', 'scheduled__2025-07-24T02:00:00+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/final_exam_src.py', '--cfg-path', '/tmp/tmpnohfq5nu']
[2025-07-24T15:35:12.393+0000] {standard_task_runner.py:91} INFO - Job 5: Subtask Stage_songs
[2025-07-24T15:35:12.834+0000] {task_command.py:426} INFO - Running <TaskInstance: tunestream_etl_pipeline.Stage_songs scheduled__2025-07-24T02:00:00+00:00 [running]> on host 562a5fa0e436
[2025-07-24T15:35:13.833+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='tunestream_etl_pipeline' AIRFLOW_CTX_TASK_ID='Stage_songs' AIRFLOW_CTX_EXECUTION_DATE='2025-07-24T02:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-24T02:00:00+00:00'
[2025-07-24T15:35:13.858+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-24T15:35:14.206+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-24T15:35:14.267+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/final_exam_src.py", line 79, in load_songs_into_table
    postgres_hook.run(insert_sql, parameters=(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/common/sql/hooks/sql.py", line 431, in run
    with self._create_autocommit_connection(autocommit) as conn:
  File "/usr/local/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/common/sql/hooks/sql.py", line 551, in _create_autocommit_connection
    with closing(self.get_conn()) as conn:
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/postgres/hooks/postgres.py", line 145, in get_conn
    conn = deepcopy(self.connection or self.get_connection(conn_id))
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/hooks/base.py", line 83, in get_connection
    conn = Connection.get_connection_from_secrets(conn_id)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/connection.py", line 519, in get_connection_from_secrets
    raise AirflowNotFoundException(f"The conn_id `{conn_id}` isn't defined")
airflow.exceptions.AirflowNotFoundException: The conn_id `postgres_default` isn't defined
[2025-07-24T15:35:14.333+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=tunestream_etl_pipeline, task_id=Stage_songs, run_id=scheduled__2025-07-24T02:00:00+00:00, execution_date=20250724T020000, start_date=20250724T153512, end_date=20250724T153514
[2025-07-24T15:35:14.447+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 5 for task Stage_songs (The conn_id `postgres_default` isn't defined; 106)
[2025-07-24T15:35:14.560+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-24T15:35:14.631+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.10/site-packages/***/models/baseoperator.py:1297: AirflowProviderDeprecationWarning: Call to deprecated class PostgresOperator. (Please use `***.providers.common.sql.operators.sql.SQLExecuteQueryOperator`.Also, you can provide `hook_params={'schema': <database>}`.)
  result = cls.__new__(cls)

[2025-07-24T15:35:14.705+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-24T15:35:14.754+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
